### PR TITLE
Add profile entitlements and addresses

### PR DESCRIPTION
**Pull request by GiuliaTeggi**
### Ticket:

- [14975](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/14975)

### Description

- add entitlements cards and details pages
- add new address, remove address edit address pages
- add not found page
- remove cancel buttons
- migrations to update the entitlements and addresses tables

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Dev change**

### Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works

### Screenshots:

N/A

### CODE REVIEW

File: apps/profile/app/[locale]/addresses/new-address/AddressForm.tsx [hunk 1]
```diff
@@ -0,0 +1,409 @@
+import { NextPageProps } from "../../../../types";
```
Comment (GiuliaTeggi) [inline apps/profile/app/[locale]/addresses/new-address/AddressForm.tsx:1]: This is largely copied from the `web` app `AddressForm` component

File: apps/profile/app/utils/index.ts [hunk 2]
```diff
@@ -1,2 +1,9 @@
+import dayjs from "dayjs";
+
 export * as postgres from "./postgres";
 export * as form from "./form";
+export * as routes from "./routes";
+
+export function formatDate(date: string | Date) {
```
Comment (Thurfjell) [inline apps/profile/app/utils/index.ts:7]: What about something like `formatDateDDMMYYYY()`? Super clear! :joy:

File: apps/profile/messages/en.json [hunk 3]
```diff
@@ -58,5 +85,61 @@
...
+    "selectDatesFormTitle": "Select your move in and move out dates",
+    "moveInDate": "Move in date",
+    "moveOutDate": "Move out date",
+    "day": "Day",
+    "month": "Month",
+    "year": "Year"
+  },
+  "NotFound": {
+    "title": "Not Found",
+    "description": " Sorry, the requested resource could not be found",
```
Comment (Thurfjell) [inline apps/profile/messages/en.json:119]: space here intentional?

**Review by Thurfjell — APPROVED** [review]

File: apps/profile/app/components/AboutMe.tsx [hunk 4]
```diff
@@ -5,6 +5,10 @@ import Addresses from "./Addresses";
 import Entitlements from "./Entitlements";
 import Consent from "./Consent";
 
+const Line = () => (
+  <hr style={{ marginBottom: "30px", color: ds.colours.ogcio.midGrey }} />
```
[the remark this answers is not in the document]
Comment (GiuliaTeggi) [reply apps/profile/app/components/AboutMe.tsx:9]: Yes found it! Thanks, will replace

File: apps/profile/messages/en.json (hunk 3 above)
Comment (GiuliaTeggi) [reply apps/profile/messages/en.json:119]: Nope

File: apps/profile/app/utils/index.ts (hunk 2 above)
Comment (GiuliaTeggi) [reply apps/profile/app/utils/index.ts:7]: This is the only way we format dates on profile, so I'll leave that to whenever we need other types of date formatting :D

**Pull request merged**